zipline/pipeline/loaders/events.py

```diff
@@ -36,196 +42,128 @@
 
 class EventsLoader(PipelineLoader):
     """
-    Abstract loader.
+    Base class for PipelineLoaders that supports loading the next and previous
+    value of an event field.
 
-    Does not currently support adjustments to the dates of known events.
+    Does not currently support adjustments.
 
     Parameters
     ----------
+    events : pd.DataFrame
+        A DatetimeIndexed DataFrame representing events (e.g. share buybacks or
+        earnings announcements) associated with particular companies.
+
+        ``events`` must contain at least three columns::
+            sid : int64
+                The asset id associated with each event.
+
+            event_date : datetime64[ns]
+                The date on which the event occurred.
+
+            timestamp : datetime64[ns]
+                The date on which we learned about the event. Must be greater
+                than or equal to the frame index value.
+
     all_dates : pd.DatetimeIndex
         Index of dates for which we can serve queries.
-    events_by_sid : dict[int -> pd.DataFrame or pd.Series or pd.DatetimeIndex]
-        Dict mapping sids to objects representing dates on which earnings
-        occurred.
-
-        If a dict value is a Series, it's interpreted as a mapping from the
-        date on which we learned an announcement was coming to the date on
-        which the announcement was made.
-
-        If a dict value is a DatetimeIndex, it's interpreted as just containing
-        the dates that announcements were made, and we assume we knew about the
-        announcement on all prior dates.  This mode is only supported if
-        ``infer_timestamp`` is explicitly passed as a truthy value.
-        Dict mapping sids to DataFrames, Series, or DatetimeIndexes.
-
-        If the value is a DataFrame, it then represents dates on which events
-        occurred along with other associated values. If the DataFrame
-        contains a "timestamp" column, that column is interpreted as the date
-        on which we learned about the event. If the DataFrames do not contain a
-         "timestamp" column, we assume we knew about the event on all prior
-         dates.  This mode is only supported if ``infer_timestamp`` is
-         explicitly passed as a truthy value.
-
-    infer_timestamps : bool, optional
-        Whether to allow omitting the "timestamp" column.
+
```

Comment:
Need to update docs to remove `all_dates` and `dataset` and include `next_value_columns` and `previous_value_columns`.